JesusFilm/core
# feat: journeys search bar

# Description

### Issue

We want our journeys page on the resources hub to use algolia for searching.

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/38213804/todos/7581223703)

### Solution

Refactor the template gallery and related components to allow us to display journeys from algolia API instead of core

# External Changes

* Add algolia hooks to related components
* Add component tests and mocks
* Comment out failing tests to do with category & language select dropdowns
* Conditionally render the algolia <Index> element so that component tests don't fail
* Add algolia data for tests
* Duplicate the watch config into shared space (needed because template components are in journeys-ui) 

# Additional information

## How to test
* As a user I must be able to see the same journey templates listed as before from NS templates page
* As a user I should be able to search by title/description to filter the journeys shown
* Grouping of template sections should match the behaviour of journeys admin
  - No search or tags = featured & new (if more than 5 journeys in a category then show category)
  - Search with no tags = featured & new (if more than 5 journeys in a category then show category)
